### DISCUSSION TITLE
The "next export" command has been removed in favor of "output: export" in next.config.js

### DISCUSSION THREAD

**Discussion by LuciferDIot** [vercel/next.js#58790] [Help] [answered]
### Summary

I was deploying in GitHub pages my Next.js application, which used threejs. I was getting this error again and again. So, I want to know what is the reason for this error and can someone guide me?
![image_2023-11-23_011409626](https://github.com/vercel/next.js/assets/105873753/b2de2979-1e0f-45e4-a92f-d3bb937b7905)

### Additional information

```js
My repository link: https://github.com/LuciferDIot/3D-Portfolio.git
```

**Comment by RenatMardanov** [comment]
You can delete these 2 lines in nextjs.yml
- name: Static HTML export with Next.js
  run: ${{ steps.detect-package-manager.outputs.runner }} next export
  
and then update next.config.js as in the documentation at the link

**Reply by LuciferDIot** [reply]
Thank you for replying to me. Now it solved, but now it's getting a new error. Can you help me with this? 
![image_2023-12-05_134020185](https://github.com/vercel/next.js/assets/105873753/6d5120e0-0791-4b7d-88ea-825e6729e8ec)

**Reply by jawney-anshita** [reply]
You need to add " output: 'export'," to the next.config.js file and then build

/** @type {import('next').NextConfig} */
const nextConfig = {
    output: 'export',
}

module.exports = nextConfig

**Reply by Pitchayut586** [reply]
thx bro

**Reply by roudlek** [reply]
Our hero

**Reply by voidmain443** [reply]
<img width="935" alt="스크린샷 2024-01-30 오후 10 14 42" src="https://github.com/vercel/next.js/assets/83549147/59df384b-91d6-49ba-a08a-73c1b2c497a2">


i had a same problam with the building workflow 

i changed `next.config.mjs` file 

```
/** @type {import('next').NextConfig} */
const nextConfig = {
  output:'export',
};
/* module.exports = nextConfig*/
export default nextConfig;

```

**Reply by julianesp** [reply]
Gracias. Se solucionó el error 👍

**Reply by jigneshloh** [reply]
your answer helped me.

**Reply by awaisYusuf** [reply]
unable to find nextjs.yml file. where that file is in next js project?

**Reply by eliasgb2409** [reply]
The nextjs.yml file is available in the .github directory that is created automatically when using GitHub Actions for hosting your Next.js project with Github Pages.

**Comment by MShoaibCodes** [comment]
Where can I find nextjs.yml?

**Reply by LuciferDIot** [reply]
Thank you for replying to me. Now it solved, but now it's getting a new error. Can you help me with this? 
![Screenshot 2023-12-05 133911](https://github.com/vercel/next.js/assets/105873753/e1b13b1f-ff8f-4c23-9289-f16e5c350d8c)

**Reply by MShoaibCodes** [reply]
Can you show me your workflow file. Its issue with a command in your workflow file

**Reply by LuciferDIot** [reply]
Below is the code

`# Sample workflow for building and deploying a Next.js site to GitHub Pages
#
# To get started with Next.js see: https://nextjs.org/docs/getting-started
#
name: Deploy Next.js site to Pages

on:
  # Runs on pushes targeting the default branch
  push:
    branches: ["main"]

  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
permissions:
  contents: read
  pages: write
  id-token: write

# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
concurrency:
  group: "pages"
  cancel-in-progress: false

jobs:
  # Build job
  build:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v4
      - name: Detect package manager
        id: detect-package-manager
        run: |
          if [ -f "${{ github.workspace }}/yarn.lock" ]; then
            echo "manager=yarn" >> $GITHUB_OUTPUT
            echo "command=install" >> $GITHUB_OUTPUT
            echo "runner=yarn" >> $GITHUB_OUTPUT
            exit 0
          elif [ -f "${{ github.workspace }}/package.json" ]; then
            echo "manager=npm" >> $GITHUB_OUTPUT
            echo "command=ci" >> $GITHUB_OUTPUT
            echo "runner=npx --no-install" >> $GITHUB_OUTPUT
            exit 0
          else
            echo "Unable to determine package manager"
            exit 1
          fi
      - name: Setup Node
        uses: actions/setup-node@v4
        with:
          node-version: "20"
          cache: ${{ steps.detect-package-manager.outputs.manager }}
      - name: Setup Pages
        uses: actions/configure-pages@v4
        with:
          # Automatically inject basePath in your Next.js configuration file and disable
          # server side image optimization (https://nextjs.org/docs/api-reference/next/image#unoptimized).
          #
          # You may remove this line if you want to manage the configuration yourself.
          static_site_generator: next
      - name: Restore cache
        uses: actions/cache@v3
        with:
          path: |
            .next/cache
          # Generate a new cache whenever packages or source files change.
          key: ${{ runner.os }}-nextjs-${{ hashFiles('**/package-lock.json', '**/yarn.lock') }}-${{ hashFiles('**.[jt]s', '**.[jt]sx') }}
          # If source files changed but packages didn't, rebuild from a prior cache.
          restore-keys: |
            ${{ runner.os }}-nextjs-${{ hashFiles('**/package-lock.json', '**/yarn.lock') }}-
      - name: Install dependencies
        run: ${{ steps.detect-package-manager.outputs.manager }} ${{ steps.detect-package-manager.outputs.command }}
      - name: Build with Next.js
        run: ${{ steps.detect-package-manager.outputs.runner }} next build
      - name: Upload artifact
        uses: actions/upload-pages-artifact@v2
        with:
          path: ./out

  # Deployment job
  deploy:
    environment:
      name: github-pages
      url: ${{ steps.deployment.outputs.page_url }}
    runs-on: ubuntu-latest
    needs: build
    steps:
      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@v3`

**Reply by JosemyDuarte** [reply]
Could it be that you are using a `distDir` different from `out`? Looks like the upload artifact is failing to find `out`. Maybe you are using `dist`. If that's the case you need to update the `out` to `dist` here:

```
uses: actions/upload-pages-artifact@v2
with:
path: ./out
```

**Reply by josealvis** [reply]
If you are still having the problem be sure you do not have this line: 
static_site_generator: next
This line overwrites your next.config file.

**Reply by jigneshloh** [reply, marked as answer]
I was facing the same issue.  commented cmd "static_site_generator: next" in deployment file nextjs.yml 
Solved my issues.

**Reply by md-fahad-ali** [reply]
still not work after commenting this one

**Reply by davidde** [reply]
You also have to comment out the `with:` header of `static_site_generator: next`, because an empty field for `with` is invalid.

**Comment by 5267lq** [comment]
Maybe we deploy it on window platform but it's for ubuntu system. I meat the same problem.

**Reply by MShoaibCodes** [reply]
I tried deploying in an ubuntu server. Still the same issue occurred

**Reply by Anth0nyQ** [reply]
@MShoaibCodes Please post your workflow file. I found that if your project is structured in a certain way you should put full paths and it works out.

**Reply by MShoaibCodes** [reply]
on:
  push:
    branches:
      - main

name: 🚀 Deploy website on push
jobs:
  web-deploy:
    runs-on: ubuntu-latest
    name: 🎉 Deploy
    steps:
      - name: 🚚 Get latest code
        uses: actions/checkout@v3

      - name: Use Node.js 18
        uses: actions/setup-node@v2
        with:
          node-version: "18"
          
      - name: 🔨 Build Project
        run: |
          npm install
          npm run export
          ls
          
      - name: 📂 Sync files
        uses: samKirkland/FTP-Deploy-Action@4.0.0
        with:
          server: ${{ secrets.HOST }}
          username: ${{ secrets.FTP_USERNAME }}
          password: ${{ secrets.FTP_PASSWORD }}
          port: ${{ secrets.PORT }}

**Reply by MitoruDev** [reply]
![image](https://github.com/vercel/next.js/assets/123580397/3c12de45-e22c-427d-a2ad-b43a101d70d0)
I have the same error

`# Sample workflow for building and deploying a Next.js site to GitHub Pages
#
# To get started with Next.js see: https://nextjs.org/docs/getting-started
#
name: Deploy Next.js site to Pages

on:
  # Runs on pushes targeting the default branch
  push:
    branches: ["main"]

  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
permissions:
  contents: read
  pages: write
  id-token: write

# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
concurrency:
  group: "pages"
  cancel-in-progress: false

jobs:
  # Build job
  build:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v4
      - name: Detect package manager
        id: detect-package-manager
        run: |
          if [ -f "${{ github.workspace }}/yarn.lock" ]; then
            echo "manager=yarn" >> $GITHUB_OUTPUT
            echo "command=install" >> $GITHUB_OUTPUT
            echo "runner=yarn" >> $GITHUB_OUTPUT
            exit 0
          elif [ -f "${{ github.workspace }}/package.json" ]; then
            echo "manager=npm" >> $GITHUB_OUTPUT
            echo "command=ci" >> $GITHUB_OUTPUT
            echo "runner=npx --no-install" >> $GITHUB_OUTPUT
            exit 0
          else
            echo "Unable to determine package manager"
            exit 1
          fi
      - name: Setup Node
        uses: actions/setup-node@v4
        with:
          node-version: "20"
          cache: ${{ steps.detect-package-manager.outputs.manager }}
      - name: Setup Pages
        uses: actions/configure-pages@v4
        with:
          # Automatically inject basePath in your Next.js configuration file and disable
          # server side image optimization (https://nextjs.org/docs/api-reference/next/image#unoptimized).
          static_site_generator: next
      - name: Restore cache
        uses: actions/cache@v4
        with:
          path: |
            .next/cache
          # Generate a new cache whenever packages or source files change.
          key: ${{ runner.os }}-nextjs-${{ hashFiles('**/package-lock.json', '**/yarn.lock') }}-${{ hashFiles('**.[jt]s', '**.[jt]sx') }}
          # If source files changed but packages didn't, rebuild from a prior cache.
          restore-keys: |
            ${{ runner.os }}-nextjs-${{ hashFiles('**/package-lock.json', '**/yarn.lock') }}-
      - name: Install dependencies
        run: ${{ steps.detect-package-manager.outputs.manager }} ${{ steps.detect-package-manager.outputs.command }}
      - name: Build with Next.js
        run: ${{ steps.detect-package-manager.outputs.runner }} next build
      - name: Upload artifact
        uses: actions/upload-pages-artifact@v3
        with:
          path: ./dist

  # Deployment job
  deploy:
    environment:
      name: github-pages
      url: ${{ steps.deployment.outputs.page_url }}
    runs-on: ubuntu-latest
    needs: build
    steps:
      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@v4`

**Reply by oguzalpakgul** [reply]
I had the same error. it was fixed when I put the following line in the comment line.

```
- name: Setup Pages
uses: actions/configure-pages@v4
with:
# Automatically inject basePath in your Next.js configuration file and disable
# server side image optimization (https://nextjs.org/docs/api-reference/next/image#unoptimized).
static_site_generator: next
```

The last part `path: ./dist` is the file name at the output point. If you run npm run build on your computer and it outputs to the "./out" file, you need to put `path: ./out` in the path.

**Comment by Mgphone** [comment]
```
`# Sample workflow for building and deploying a Next.js site to GitHub Pages
#
# To get started with Next.js see: https://nextjs.org/docs/getting-started
#
name: Deploy Next.js site to Pages

on:

#Runs on pushes targeting the default branch
push:
branches: ["main"]

#Allows you to run this workflow manually from the Actions tab
workflow_dispatch:

#Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
permissions:
contents: read
pages: write
id-token: write

#Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
#However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
concurrency:
group: "pages"
cancel-in-progress: false

jobs:

#Build job
build:
runs-on: ubuntu-latest
steps:
- name: Checkout
uses: actions/checkout@v4
- name: Detect package manager
id: detect-package-manager
run: |
if [ -f "${{ github.workspace }}/yarn.lock" ]; then
echo "manager=yarn" >> $GITHUB_OUTPUT
echo "command=install" >> $GITHUB_OUTPUT
echo "runner=yarn" >> $GITHUB_OUTPUT
exit 0
elif [ -f "${{ github.workspace }}/package.json" ]; then
echo "manager=npm" >> $GITHUB_OUTPUT
echo "command=ci" >> $GITHUB_OUTPUT
echo "runner=npx --no-install" >> $GITHUB_OUTPUT
exit 0
else
echo "Unable to determine package manager"
exit 1
fi
- name: Setup Node
uses: actions/setup-node@v4
with:
node-version: "20"
cache: ${{ steps.detect-package-manager.outputs.manager }}
- name: Setup Pages
uses: actions/configure-pages@v4
with:
# Automatically inject basePath in your Next.js configuration file and disable
# server side image optimization (https://nextjs.org/docs/api-reference/next/image#unoptimized).
#
# You may remove this line if you want to manage the configuration yourself.
static_site_generator: next
- name: Restore cache
uses: actions/cache@v3
with:
path: |
.next/cache
# Generate a new cache whenever packages or source files change.
key: ${{ runner.os }}-nextjs-${{ hashFiles('/package-lock.json', '/yarn.lock') }}-${{ hashFiles('.[jt]s', '.[jt]sx') }}
# If source files changed but packages didn't, rebuild from a prior cache.
restore-keys: |
${{ runner.os }}-nextjs-${{ hashFiles('/package-lock.json', '/yarn.lock') }}-
- name: Install dependencies
run: ${{ steps.detect-package-manager.outputs.manager }} ${{ steps.detect-package-manager.outputs.command }}
- name: Build with Next.js
run: ${{ steps.detect-package-manager.outputs.runner }} next build
- name: Upload artifact
uses: actions/upload-pages-artifact@v2
with:
path: ./dist

#Deployment job
deploy:
environment:
name: github-pages
url: ${{ steps.deployment.outputs.page_url }}
runs-on: ubuntu-latest
needs: build
steps:
- name: Deploy to GitHub Pages
id: deployment
uses: actions/deploy-pages@v3`

```
> 
i still got an error of 
<img width="461" alt="errror" src="https://github.com/vercel/next.js/assets/51868037/91fef8f6-bf43-4d15-97b5-456253b88262">

**Reply by bensontrent** [reply]
Remove `next export` from your package.json then update your  next.config file with the export option as described here:  https://nextjs.org/docs/app/building-your-application/deploying/static-exports
